scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/task/dataspaces/RemoteSpace.java

```diff
@@ -115,16 +150,18 @@
      *  * matches zero or more characters<br>
      *  ? matches one character
      *
+     * @param remotePath path in the RemoteSpace used as base for the pattern (e.g. "/")
      * @param pattern pattern to locate files inside the RemoteSpace
      * @throws FileSystemException
+     * @see "https://docs.oracle.com/javase/7/docs/api/java/nio/file/FileSystem.html#getPathMatcher(java.lang.String)"
      */
-    void deleteFiles(String pattern) throws FileSystemException;
+    void deleteFiles(String remotePath, String pattern) throws FileSystemException;
 
     /**
-     * Returns the root URL of the RemoteSpace
+     * Returns the URLs of the RemoteSpace
      * @return URL to the space
      */
-    String getSpaceURL();
+    List<String> getSpaceURLs() throws NotConnectedException, PermissionException, FileSystemException;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: org.ow2.proactive.scheduler.common.exception.NotConnectedException, org.ow2.proactive.scheduler.common.exception.PermissionException, org.ow2.proactive.scheduler.common.task.dataspaces.FileSystemException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)